var/spack/repos/builtin/packages/libxshmfence/package.py

```diff
@@ -13,11 +13,18 @@ class Libxshmfence(AutotoolsPackage, XorgPackage):
     with the X SyncFence objects that can be shared between processes
     using file descriptor passing."""
 
-    homepage = "https://cgit.freedesktop.org/xorg/lib/libxshmfence/"
-    xorg_mirror_path = "lib/libxshmfence-1.3.tar.bz2"
+    homepage = "https://gitlab.freedesktop.org/xorg/lib/libxshmfence"
+    xorg_mirror_path = "lib/libxshmfence-1.3.2.tar.xz"
+
+    # note: url_for_version can only return a single url, no mirrors
+    def url_for_version(self, version):
+        if self.spec.satisfies("@:1.3.0"):
+            return spack.url.substitute_version(self.urls[0].replace("xz", "bz2"), version)
```

Comment:
I'd ditch the note comment and make the following changes, which work for all of the currently listed versions:

```suggestion
        url = super().url_for_version(version)
        if version <= Version("1.3"):
            return url.replace("xz", "bz2")

        return url
```

Reply:
Note:  The original `self.spec.satisifes()` doesn't work in `url_for_version()` since spec is not concrete at that point.